cmd/lakectl/cmd/log.go

```diff
@@ -151,13 +169,19 @@ var logCmd = &cobra.Command{
 				},
 			}
 
+			// case --no-merges, filter commits and subtract that amount from amount desired
+			if noMerges {
+				data.Commits = filterMergeCommits(data.Commits)
+				amount = amount - (3 * len(data.Commits))
```

Comment:
Why is it tripled here?

Reply:
My thought was that if we use that heuristic, in this way we stay true to it. We asked for three times the amount we needed to get ~1/3 results of that. Updating the amount this way will keep that heuristic in every request.

Reply:
It's better if we have a single location to handle amount heuristics. Can you try to combine this block and the above one (lines 114-120) into one? Preferably where the loop starts